scripts/portfolio.js

```diff
@@ -10,15 +9,41 @@ function portfolio (data) {
   this.description = data.description;
 }
 
+portfolio.all = [];
+
 portfolio.prototype.toHtml = function() {
   var template = Handlebars.compile($('#portfolio_template').text());
   return template(this);
 };
 
-portfolioData.forEach(function(ele) {
-  portfolios.push(new portfolio(ele));
-});
+portfolio.fetchAll = function () {
+  if (localStorage.portfolioData) {
+    $.ajax({
+      type: 'HEAD',
+      url: 'data/portfolioData.json',
+      success: function(data,message,xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        if (!localStorage.eTag || eTag !== localStorage.eTag) {
+          localStorage.eTag = eTag;
+        } else {
+          portfolio.loadAll(JSON.parse(localStorage.portfolioData));
+        }
+      }
+    });
+    portfolio.loadAll(JSON.parse(localStorage.portfolioData));
+    viewer.initIndexPage();
+  } else {
+    $.getJSON('data/portfolioData.json', function(data) {
+      var stringData = JSON.stringify(data);
+      localStorage.setItem('portfolioData',stringData);
+      portfolio.loadAll(JSON.parse(localStorage.portfolioData));
+      viewer.initIndexPage();
+    });
```

Comment:
See the notes in your portfolio.js file for how to change this line.  You have the same situation here.